libraries/chain/genesis.cpp

```diff
@@ -0,0 +1,190 @@
+            auth.account = account.name;
+            auth.owner.add_authority(account.public_key, 1);
+            auth.owner.weight_threshold = 1;
+            auth.active = auth.owner;
+            auth.posting = auth.active;
+        });
+    }
+}
+
+void database::init_genesis_witnesses(const genesis_state_type& genesis_state)
+{
+    const std::vector<genesis_state_type::witness_type>& witnesses = genesis_state.witness_candidates;
+
+    for (auto& witness : witnesses)
+    {
+        FC_ASSERT(!witness.owner_name.empty(), "Witness 'owner_name' should not be empty.");
+
+        create<witness_object>([&](witness_object& w) {
+            w.owner = witness.owner_name;
+            w.signing_key = witness.block_signing_key;
+            w.schedule = witness_object::top19;
+            w.hardfork_time_vote = get_genesis_time();
+        });
+    }
+}
+
+void database::init_genesis_witness_schedule(const genesis_state_type& genesis_state)
+{
+    const std::vector<genesis_state_type::witness_type>& witness_candidates = genesis_state.witness_candidates;
+
+    create<witness_schedule_object>([&](witness_schedule_object& wso) {
+        for (size_t i = 0; i < wso.current_shuffled_witnesses.size() && i < witness_candidates.size(); ++i)
+        {
+            wso.current_shuffled_witnesses[i] = witness_candidates[i].owner_name;
+        }
+    });
+}
+
+void database::init_genesis_global_property_object(const genesis_state_type& genesis_state)
+{
+    create<dynamic_global_property_object>([&](dynamic_global_property_object& gpo) {
+        gpo.time = get_genesis_time();
+        gpo.recent_slots_filled = fc::uint128::max_value();
+        gpo.participation_count = 128;
+        gpo.current_supply = asset(genesis_state.init_supply, SCORUM_SYMBOL);
+        gpo.maximum_block_size = SCORUM_MAX_BLOCK_SIZE;
+
+        gpo.total_reward_fund_scorum = asset(0, SCORUM_SYMBOL);
+        gpo.total_reward_shares2 = 0;
+    });
+}
+
+void database::init_genesis_rewards(const genesis_state_type& genesis_state)
+{
+    const auto& gpo = get_dynamic_global_properties();
+
+    auto post_rf = create<reward_fund_object>([&](reward_fund_object& rfo) {
+        rfo.name = SCORUM_POST_REWARD_FUND_NAME;
+        rfo.last_update = head_block_time();
+        rfo.percent_curation_rewards = SCORUM_1_PERCENT * 25;
+        rfo.percent_content_rewards = SCORUM_100_PERCENT;
+        rfo.reward_balance = gpo.total_reward_fund_scorum;
+        rfo.author_reward_curve = curve_id::linear;
+        rfo.curation_reward_curve = curve_id::square_root;
+    });
+
+    // As a shortcut in payout processing, we use the id as an array index.
+    // The IDs must be assigned this way. The assertion is a dummy check to ensure this happens.
+    FC_ASSERT(post_rf.id._id == 0);
+
+    // clang-format off
+    asset initial_pool_supply(genesis_state.init_rewards_supply.amount * SCORUM_GUARANTED_REWARD_SUPPLY_PERIOD_IN_DAYS / SCORUM_REWARDS_INITIAL_SUPPLY_PERIOD_IN_DAYS);
```

Comment:
Add asset symbol to initial_pool_supply. It is not obvious if u use symbol from default constructor

Reply:
OK, will change